packages/ocapn-value/src/toCBOR.js

```diff
@@ -0,0 +1,65 @@
+// @ts-check
+import cbor from "cbor";
+import { MT } from "cbor/types/lib/constants";
+import { tagSymbol } from "./ocapn-value";
+
+// https://www.rfc-editor.org/rfc/rfc8949.html#section-3.4.6
+const SELF_DESCRIBED = 55799;
+
+/**
+ * TODO: Cap
+ * TODO: check return value from encoder methods
+ *
+ * @param {cbor.Encoder} encoder
+ * @param {import('./ocapn-value').Value<never, Error>} value
+ */
+export const toCBOR = (encoder, value) => {
+  switch (typeof value) {
+    case "undefined":
+    case "boolean":
+    case "number":
+    case "string":
+    case "bigint":
+      encoder.pushAny(value);
+      break;
+    case "symbol":
+      encoder._pushTag(SELF_DESCRIBED);
+      encoder.pushAny("S");
+      encoder.pushAny(value.toString());
+      break;
+    case "object":
+      if (value === null) {
+        encoder.push(null);
+      } else if (value instanceof Error) {
+        encoder._pushTag(SELF_DESCRIBED);
+        encoder.pushAny("E");
+        encoder._pushInt(2, MT.ARRAY);
+        encoder.pushAny(value.name);
+        encoder.pushAny(value.message);
```

Comment:
likewise, this SELF_DESCRIBED should be followed by 1 array of 3 items: "E" tag, name, message.